### PR TITLE
[SDPA-5894] Replace unicode spaces

### PR DESCRIPTION
**Pull request by waitingallday**
## Motivation and Context

~**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPA-5756~
**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPA-5894

## Changed

1. Added story to demonstrate unicode spaces and example unicode replacement plugin on `RplMarkup`
2. Added unicode replacement plugin to `packages/ripple-nuxt-tide/lib/config/markup-plugins.js`
3. Updated fixture for `packages/ripple-nuxt-tide/test/unit/markup-transpiler.test.js` and related snapshot

### Screenshots

The story at `/?path=/story/organisms-markup--unicode-spaces` has been created with each different unicode space inserted after the first word of each callout box. eg.

Screen Shot 2021-12-02 at 5 10 39 pm

If the replacement worked, it should just look like a normal space. The first one (screen above) is a good example, since the *zero width space* should render as _no_ space on a normal page. Here, it's being replaced with an actual space.

## How Has This Been Tested?

Updated the fixture for `packages/ripple-nuxt-tide/test/unit/markup-transpiler.test.js` to include a block demonstrating all 19 unicode space characters. Snapshot shows only normal spaces.

Disabling the plugin then fails the snapshot test.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist

- [x] I've added relevant changes to the documentation.
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [ ] My change requires a template update for create-ripple-app.
- [ ] I have added template update script for next release.

### CODE REVIEW

File: packages/components/Organisms/Markup/examplePlugins.js [hunk 1]
```diff
@@ -164,12 +165,23 @@ const pluginLinks = function () {
   })
 }
 
+const pluginReplaceUnicodeWhitespace = function () {
+  // Match text nodes only
+  this.find('*').map((i, n) => n.children).filter((i, n) => n.type === 'text').map((i, node) => {
```
Comment (dylankelly) [inline packages/components/Organisms/Markup/examplePlugins.js:170]: discussed formatting here as per line, might make it a bit more readable?

File: packages/ripple-nuxt-tide/lib/config/markup-plugins.js [hunk 2]
```diff
@@ -5,6 +5,7 @@
 // See a example in `pluginEmbeddedMediaVideo` plugin below.
 
 import { getAnchorLinkName } from '@dpc-sdp/ripple-global/utils/helpers.js'
+import CodepointMap from '@dpc-sdp/ripple-global/utils/codepoint.map.json'
```
Comment (dylankelly) [inline packages/ripple-nuxt-tide/lib/config/markup-plugins.js:8]: should be camelCase instead of PascalCase ?

**Review by dylankelly — APPROVED** [review]
Nice one @waitingallday!  some trivial things I'm happy to leave

**Comment by waitingallday** [comment]
Arghh I forgot those changes, I'll push it up now!

**Review by dylankelly — APPROVED** [review]
👍

**Pull request merged**